### PR TITLE
Elijah/docker dev env

### PR DESCRIPTION
**Pull request by ElijahCaine**
This now uses docker-compose, which gives us a few advantages:
- The dev environment can be easily extended.
- The setup/management of the dev envs doesn't depend on a (*shudder*) **shell script**.
- If we choose to host this on a machine using docker it *should* be easy to deploy with -- you guessed it: `docker-compose`!
@Kennric @matthewrsj take a look when you get a chance.

**note** Requires `docker >= 1.10.x` and `docker-compose >= 1.6.x`. Check yo' dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,48 @@
...
+```
+$ pip install --user docker-compose
+```
+
+**IMPORTANT**: Next copy dockerfiles/app.dist to dockerfiles/app and edit the
+variables:
+
+```
+ENV USER     you_user_name
+ENV PASSWORD you_container_password
```
Comment (matthewrsj) [inline README.md]: `your_container_password`
......^

File: README.md [hunk 2]
```diff
@@ -1,15 +1,48 @@
...
+ENV UID      your_local_uid
+```
+
+Finally, run the `docker-compose` commands you need to actually start
+developing.
+
+```
+$ docker-compose up dev   # starts running the app on port 8000
+# ~~~~or~~~~
+$ docker-compose run --service-ports dev bash   # puts you in a dev shell
```
Comment (matthewrsj) [inline README.md]: These commands only work for me when I `sudo ...` even though my user is added to the `docker` group. This may just be a local issue...

**Comment by matthewrsj** [comment]
I get a few warnings during the `docker-compose up dev` run:

```
...
warning: /var/cache/yum/x86_64/7/updates/packages/bind-license-9.9.4-29.el7_2.3.noarch.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
...
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/sinatra/images/404.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/sinatra/images/500.png, skipping
...
useradd: warning: the home directory already exists.
Not copying any file from skel directory into it.
...
```

but everything works when the run completes, so I'm not too worried about it. Other than the 1.5 comments above, this looks good.

**Comment by matthewrsj** [comment]
You have at least 6 commits from another PR here.

**Comment by matthewrsj** [comment]
Yay, it works! I'll +1 after you rebase.

**Comment by matthewrsj** [comment]
This works great and it is well documented. +1

**Comment by matthewrsj** [comment]
Still works, and not even any errors! :+1:

**Comment by ElijahCaine** [comment]
@Kennric @alxngyn any reason we haven't merged this yet?

**Comment by alxngyn** [comment]
+1 from me 
Sorry for the delay.

**Pull request merged**